test/Microsoft.ML.Tests/Transformers/TextFeaturizerTests.cs

```diff
@@ -51,6 +52,95 @@ public void TextFeaturizerWithPredefinedStopWordRemoverTest()
+                Norm = TextFeaturizingEstimator.NormFunction.None,
+                OutputTokensColumnName = "OutputTokens"
+            };
+            var pipeline = ML.Transforms.Text.FeaturizeText("Features", options, "A");
+            var model = pipeline.Fit(dataView);
+            var engine = model.CreatePredictionEngine<TestClass, TestClass>(ML);
+
+            var prediction = engine.Predict(data[0]);
+            Assert.Equal("this is some text in english", string.Join(" ", prediction.OutputTokens));
+            var expected = new float[] { 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 0.0f, 0.0f };
+            Assert.Equal(expected, prediction.Features);
+
+            prediction = engine.Predict(data[1]);
+            Assert.Equal("this is another example", string.Join(" ", prediction.OutputTokens));
+            expected = new float[] { 1.0f, 1.0f, 0.0f, 0.0f, 0.0f, 0.0f, 1.0f, 1.0f };
+            Assert.Equal(expected, prediction.Features);
+        }
+
+        [Fact]
+        public void TextFeaturizerWithCharFeatureExtractorTest()
+        {
+            var data = new[] { new TestClass() { A = "abc efg", OutputTokens=null},
+                               new TestClass() { A = "xyz", OutputTokens=null } };
+            var dataView = ML.Data.LoadFromEnumerable(data);
+
+            var options = new TextFeaturizingEstimator.Options()
+            {
+                WordFeatureExtractor = null,
+                CharFeatureExtractor = new WordBagEstimator.Options() { NgramLength = 1 },
+                Norm = TextFeaturizingEstimator.NormFunction.None,
+                OutputTokensColumnName = "OutputTokens"
+            };
+            var pipeline = ML.Transforms.Text.FeaturizeText("Features", options, "A");
+            var model = pipeline.Fit(dataView);
+            var engine = model.CreatePredictionEngine<TestClass, TestClass>(ML);
+
+            var prediction = engine.Predict(data[0]);
+            Assert.Equal("abc efg", string.Join(" ", prediction.OutputTokens));
+            var expected = new float[] { 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 0.0f, 0.0f, 0.0f };
+            Assert.Equal(expected, prediction.Features);
+
+            prediction = engine.Predict(data[1]);
+            Assert.Equal("xyz", string.Join(" ", prediction.OutputTokens));
+            expected = new float[] { 1.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 1.0f, 1.0f, 1.0f, 1.0f };
+            Assert.Equal(expected, prediction.Features);
+        }
+
+        [Fact]
+        public void TextFeaturizerWithL2NormTest()
+        {
+            var data = new[] { new TestClass() { A = "abc xyz", OutputTokens=null},
+                               new TestClass() { A = "xyz", OutputTokens=null } };
+            var dataView = ML.Data.LoadFromEnumerable(data);
+
+            var options = new TextFeaturizingEstimator.Options()
+            {
+                CharFeatureExtractor = new WordBagEstimator.Options() { NgramLength = 1},
```

Comment:
>CharFeatureExtractor = new WordBagEstimator.Options()  [](start = 16, length = 54)

Is this correct? it doesn't read right to initialize a CharExtractor with the options of a WordBagEstimator...

Reply:
Nice catch! I have added this as a part of issue in [https://github.com/dotnet/machinelearning/issues/2895](https://github.com/dotnet/machinelearning/issues/2895).

---
In reply to: [267968148](https://github.com/dotnet/machinelearning/pull/3036#discussion_r267968148) [](ancestors = 267968148)